src/main/scala/uk/gov/nationalarchives/referencegenerator/DynamoDb.scala

```diff
@@ -0,0 +1,68 @@
+package uk.gov.nationalarchives.referencegenerator
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.Logger
+import io.circe.syntax._
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, ReturnValue, UpdateItemRequest}
+import uk.gov.nationalarchives.referencegenerator.Reference._
+
+import java.util
+import scala.jdk.CollectionConverters.{MapHasAsJava, MapHasAsScala}
+import scala.util.Try
+
+class DynamoDb(ddb: DynamoDbClient) {
+  val logger: Logger = Logger[DynamoDb]
+  val config: Config = ConfigFactory.load()
+
+  val tableName: String = config.getString("tableName")
+  val key: String = config.getString("key")
+  val keyVal: String = config.getString("keyVal")
+  val pieceCounter: String = config.getString("pieceCounter")
+  val keyToGet: util.Map[String, AttributeValue] = Map(key -> AttributeValue.builder().s(keyVal).build()).asJava
+
+  val request: GetItemRequest = GetItemRequest.builder()
+    .key(keyToGet)
+    .tableName(tableName)
+    .build()
+
+  def getCount: Try[String] = {
+    Try {
+      val returnedItem: Map[String, AttributeValue] = ddb.getItem(request).item.asScala.toMap
+      if (returnedItem.nonEmpty) {
+        val keys = returnedItem
+        logger.info(s"The current counter is ${keys(pieceCounter).n()}")
+        keys(pieceCounter).n()
+      } else {
+        throw new NoSuchElementException(s"No item found with the key: $key")
+      }
+    }
+  }
+
+  def getReferences(currentCounter: String, numberOfReferences: Int): Try[String] = {
+    val request: UpdateItemRequest = UpdateItemRequest.builder()
+      .tableName(tableName)
+      .key(keyToGet)
+      .updateExpression(s"ADD $pieceCounter :incr")
+      .conditionExpression(s"$pieceCounter = :currCounter AND :incr > :zero")
+      .expressionAttributeValues(Map(
+        ":incr" -> AttributeValue.builder().n(numberOfReferences.toString).build(),
+        ":currCounter" -> AttributeValue.builder().n(currentCounter).build(),
+        ":zero" -> AttributeValue.builder().n("0").build()
+      ).asJava)
+      .returnValues(ReturnValue.ALL_NEW)
+      .build()
+
+    Try {
+      val response = ddb.updateItem(request)
```

Comment:
Should anything be done with the response for the `updateItem`? Should an error be thrown for now if it fails? See comment below about moving the actual generation of the references out of this class.